src/Components/Game/Cell.js

```diff
@@ -1,26 +1,25 @@
 import styles from "./Cell.module.css";
 
-const NewCell = (props) => {
+const Cell = (props) => {
   const givenCheck = props.value ? true : false;
-  const regex = /[^\d]/g;
 
   const blockInvalid = (event) => {
-    const blockedKeys = ['e','+','-','0'];
+    const blockedKeys = ['e','+','-','0','.'];
     if (blockedKeys.includes(event.key) || event.target.value) {
       event.preventDefault();
     }
   };
```

Comment:
Instead of keeping track of a blocklist, it will be more robust to filter out any characters not whitelisted (i.e. 0-9).

One way to achieve this is to keep track of the `<input>`'s value with `useState`. Then you could have an `onChange` prop or similar to propagate the change back up to the parent component.